Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -212,18 +212,19 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
             prevFullEvent = tupleEvent;
             prevFullEventPtr = &prevFullEvent;
         }
-        auto event = MakeEventInfo(channelId, tupleEvent, ana_setup.period, ana_setup.jet_ordering, &summary);
-        if(!ana_setup.energy_scales.count(event->GetEnergyScale())) continue;
-        if(!event->GetTriggerResults().AnyAcceptAndMatch(trigger_patterns)) continue;
+
+        analysis::EventInfoBase event = analysis::EventInfoBase(tupleEvent,EventInfoBase::GetHiggsCandidateIndex(tupleEvent,analysis::TauIdDiscriminator::byIsolationMVArun2017v2DBoldDMwLT2017),ana_setup.period,ana_setup.jet_ordering, &summary);
```

Comment:
- `EventInfoBase event(tupleEvent, ...);`
- It is better to create an additional constructor in EventInfoBase that have the second argument TauIdDiscriminator and calls the original constructor + GetHiggsCandidateIndex